test/zzz-rsyslog.sh

```diff
@@ -235,3 +240,87 @@ rpod_ip="$( oc get pod ${rpod} -o jsonpath='{.status.podIP}' )"
 
 os::cmd::try_until_success "curl -s -k https://${rpod_ip}:24231/metrics"
 curl -s -k https://${rpod_ip}:24231/metrics >> $ARTIFACT_DIR/${rpod}-metrics-scrape 2>&1 || :
+
+# Test logrotation (LOG400) - OKD 4.2 and above
+# check rsyslog logs
+rpod=$( get_running_pod rsyslog )
+oc logs $rpod -c rsyslog > $ARTIFACT_DIR/rsyslog.log 2>&1
+oc exec $rpod -c rsyslog -- ls -l /var/log/rsyslog/rsyslog.log > $ARTIFACT_DIR/rsyslog.exec.txt 2>&1
+oc exec $rpod -c rsyslog -- cat /var/log/rsyslog/rsyslog.log >> $ARTIFACT_DIR/rsyslog.exec.txt 2>&1
+os::cmd::expect_success_and_text "oc logs $rpod -c rsyslog | grep 'oc exec <pod_name> -- logs'" "^oc exec .pod_name. -- logs$"
+os::cmd::expect_success "oc exec $rpod -c rsyslog -- ls /var/log/rsyslog/rsyslog.log > /dev/null 2>&1"
+logsize=$( oc exec $rpod -c rsyslog -- wc -c /var/log/rsyslog/rsyslog.log | awk '{print $1}' )
+
+# Check if logrotate works as expected
+if [ $logsize -gt 0 ]; then
+    # set max log file count and max log size
+    maxcount=3
+    maxsize=$( expr $logsize / $maxcount )
+    oc set env daemonset/rsyslog LOGGING_FILE_SIZE=$maxsize LOGGING_FILE_AGE=$maxcount
```

Comment:
this will restart rsyslog - you should try_until_failure for `oc get pod $rpod`, then try_until_success for `oc get pods | grep 'rsyslog.*Running'` or something like that

Reply:
never mind - I see you do this below